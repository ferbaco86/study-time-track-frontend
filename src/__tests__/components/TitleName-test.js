import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/store';
import TitleName from '../../components/TitleName';

describe('Rendering component', () => {
  it('creates a TitleName component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <TitleName>
              NewTitle
            </TitleName>
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('NewTitle');
  });
});

describe('TitleName', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <TitleName>
            NewTitle
          </TitleName>
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
