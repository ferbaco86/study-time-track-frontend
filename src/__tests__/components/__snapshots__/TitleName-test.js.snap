// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleName renders correctly 1`] = `
<h1
  className="sc-bdfBwQ eiuSuU"
>
  NewTitle
</h1>
`;
